tests/test_input_damages.py

```diff
@@ -0,0 +1,1287 @@
+                                "variable": (["variable"], ["gdp"]),
+                                "demand_topcode": (["demand_topcode"], ["agshare_10"]),
+                                "Es_Ed": (["Es_Ed"], ["0.1_-0.04"]),
+                                "gcm": (["gcm"], [g]),
+                                "rcp": (["rcp"], [r]),
+                                "batch": (["batch"], [b]),
+                                "model": (["model"], [m]),
+                                "market_level": (["market_level"], ["continent"]),
+                            },
+                        )
+
+                        ds_in_save = ds_in.sel(iam=m, market_level="continent")
+
+                        ds_in_save.to_netcdf(infile)
+
+    ds_out_expected = xr.Dataset(
+        {
+            "delta": (
+                ["gcm", "model", "rcp", "ssp", "batch", "region", "year"],
+                np.float32(np.full((2, 2, 2, 2, 2, 2, 2), 2 * -1 * 1.273526)),
+            ),
+        },
+        coords={
+            "batch": (["batch"], ["batch6", "batch9"]),
+            "gcm": (["gcm"], ["ACCESS1-0", "GFDL-CM3"]),
+            "model": (["model"], ["IIASA GDP", "OECD Env-Growth"]),
+            "rcp": (["rcp"], ["rcp45", "rcp85"]),
+            "region": (["region"], ["ZWE.test_region", "USA.test_region"]),
+            "ssp": (["ssp"], ["SSP2", "SSP3"]),
+            "year": (["year"], [2010, 2099]),
+        },
+    )
+
+    compute_ag_damages(
+        input_path=os.path.join(tmp_path, "ag_in"),
+        pop=econvars_fixture.econ_vars.pop,
+        topcode="agshare_10",
+        integration=True,
+        varname="delta",
+        save_path=os.path.join(tmp_path, "ag_in", "agriculture_test_output.zarr"),
+        scalar=1,
+        batches=[6, 9],
+    )
+
+    ds_out_actual = xr.open_zarr(
+        os.path.join(tmp_path, "ag_in", "agriculture_test_output.zarr")
+    )
+
+    xr.testing.assert_equal(ds_out_expected, ds_out_actual)
+
+
+@pytest.fixture
+def energy_in_csv_fixture(tmp_path):
+    """
+    Create fake energy input data for tests and save to csv
+    """
+    rcp = ["rcp45", "rcp85"]
+    gcm = ["ACCESS1-0", "GFDL-CM3"]
+    model = ["low", "high"]
+    ssp = ["SSP2", "SSP3"]
+    batch = ["batch" + str(i) for i in range(0, 15)]
+
+    for var in ["delta", "histclim"]:
+        for r in rcp:
+            for g in gcm:
+                for m in model:
+                    for s in ssp:
+                        for b in batch:
+                            d = os.path.join(tmp_path, "energy_in_csv", b, r, g, m, s)
+                            if not os.path.exists(d):
+                                os.makedirs(d)
+                            infile = os.path.join(
+                                d, f"TINV_clim_integration_total_energy_{var}.csv"
+                            )
+                            df = {
+                                "region": [
+                                    "ZWE.test_region",
```

Comment:
did you mean to have duplicate regions?

Reply:
Yes, the input .csv file is in long format, which means there is a row for each region-year.
https://github.com/ClimateImpactLab/dscim/blob/1aaa8e286ad7e269c5110ee46634e0a98cced6c7/tests/test_input_damages.py#L485-L494